app/src/main/java/com/cornellappdev/uplift/ui/components/goalsetting/EditReminderCard.kt

```diff
@@ -0,0 +1,314 @@
+import androidx.compose.animation.core.animateDpAsState
+import androidx.compose.animation.expandVertically
+import androidx.compose.animation.shrinkVertically
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.Checkbox
+import androidx.compose.material3.CheckboxDefaults
+import androidx.compose.material3.ElevatedCard
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.cornellappdev.uplift.ui.components.general.DayOfWeekBar
+import com.cornellappdev.uplift.ui.components.general.ReminderSwitch
+import com.cornellappdev.uplift.ui.screens.Reminder
+import com.cornellappdev.uplift.util.GRAY04
+import com.cornellappdev.uplift.util.GRAY08
+import com.cornellappdev.uplift.util.PRIMARY_YELLOW
+import com.cornellappdev.uplift.util.montserratFamily
+
+/**
+ * @param reminder the reminder that is currently selected
+ * @param selectedDays the days that are currently selected
+ * @param onDaySelected callback for when the selected days change
+ * @param checkedState whether the every day checkbox is checked
+ * @param onCheckedChange callback for when the every day checkbox changes
+ * @param setTimeOn whether the time picker is visible
+ * @param onSetTimeChange callback for when the time picker visibility changes
+ * @param hour the hour selected
+ * @param minute the minute selected
+ * @param isAm whether the time is AM or PM
+ * @param onHourChange callback for when the hour changes
+ * @param onMinuteChange callback for when the minute changes
+ * @param onAmChange callback for when the AM/PM changes
+ * @param addNewReminderState whether the user is currently adding a new reminder
+ * @return component that displays the day selection, time picker, and every day checkbox
+ * @sample EditReminderCard(
+ *   reminder = Reminder("12:00 PM", setOf("M", "W", "F"), true),
+ *   selectedDays = setOf("M", "W", "F"),
+ *   onDaySelected = {},
+ *   checkedState = true,
+ *   onCheckedChange = {},
+ *   setTimeOn = true,
+ *   onSetTimeChange = {},
+ *   hour = 12,
+ *   minute = "00",
+ *   isAm = false,
+ *   onHourChange = {},
+ *   onMinuteChange = {},
+ *   onAmChange = {},
+ *   addNewReminderState = false
+ *   )
+ */
```

Comment:
holy damn! That's one really well specified function. Probably the best specification we have... did you write this all yourself? 🙃

Reply:
Lol I wish, was trying to add some documentation and github copilot autocomplete did most of the work